app/views/annotations/_annotations_overview_accordion.html.erb

```diff
@@ -0,0 +1,72 @@
+<% accordionLabel = "annotations-overview-accordion" %>
+<% accordionLabel = "student-#{accordionLabel}" if is_students_annotations %>
+<div class="accordion" id="<%= accordionLabel %>">
+
+  <% annotations_by_lecture.each_with_index do |(lecture, annotations), index| %>
+    <div class="accordion-item">
+      <% collapseLabel="annotation-overview-collapse-#{index}-#{is_students_annotations}" %>
+
+        <!-- Accordion header -->
+        <h2 class="accordion-header">
+          <button class="accordion-button collapsed" type="button" data-bs-toggle="collapse"
+              data-bs-target="#<%= collapseLabel %>"
+              aria-expanded="true" aria-controls="#<%= collapseLabel %>">
+            <%= lecture %>
+          </button>
+        </h2>
+
+        <!-- Accordion body -->
+        <div class="accordion-collapse collapse" id="<%= collapseLabel %>"
+            data-bs-parent="#<%= accordionLabel %>">
+          <div class="accordion-body">
+
+            <% annotations.each do |annotation| %>
+              <a href="<%= annotation[:link] %>" style="text-decoration: none;">
+                <!-- Card shared by students will obtain their border color -->
+                <!-- according to their category via JS. -->
+                <% border_color_style = is_students_annotations ? \
+                  "" : "border-color: #{annotation[:color]}" %>
```

Comment:
This could be made into a helper method in `annotations_helper.rb` in order to get the business logic out of the view.

Reply:
I wouldn't consider this "business logic", but still outsourced this to a helper method now.